kumascript2/src/environment.js

```diff
@@ -0,0 +1,1107 @@
+/**
```

Comment:
This is the longest file in the src/ directory. Much of it is ported from the old auto-required macro files, which is why some of the code is kind of funky.